docs/tutorials/customValidationModule.md

```diff
@@ -0,0 +1,286 @@
+# Custom Validation Module
+
+In this tutorial we will look at how to create your own validation module.
+
+As mentioned in the [overview](./../modules#how-to-create-a-custom-validation-module), any custom validation module needs to extend the abstract contract [BaseAuthorizationModule](https://github.com/bcnmy/scw-contracts/blob/main/contracts/smart-account/modules/BaseAuthorizationModule.sol). This contract is crucial as it connects your module to essential authorization functions.
+The following implementation enables a module to receive userOp data and provide validation results back to the SmartAccount.
+
+Lets take a look at the step by step guide to create a custom Validation Module.
+
+**1. Create the custom validation contract**
+
+```typescript
+import {BaseAuthorizationModule} from "./BaseAuthorizationModule.sol";
+
+contract MyCustomValidationModule is BaseAuthorizationModule {
+    string public constant NAME = "My Custom Validation Module";
+    string public constant VERSION = "0.0.1";
+
+    // Function signature of this method will be used as moduleSetupData in Account Factory
+    function initForSmartAccount() {
+      
+    }
+}
+```
+
+As per the code
+
+- First we import the BaseAuthorizationModule and initialize two constants name and version
+- Then we declare a method initForSmartAccount which gets used to initialize this module for the smart account. SA calls this function with msg.sender as the smart account address. It updates the relevant storage for the msg sender, It could be ownership information as shown in ECDSAOwnsership Module.
+
+**2. Implement validateUserOp method**
```

Comment:
```suggestion
As per the code

- First, we import the BaseAuthorizationModule and initialize two constants name and version
- Then we declare a method initForSmartAccount which gets used to initialize this module for the smart account. SA calls this function with msg.sender as the smart account address. It updates the relevant storage for the msg sender, It could be ownership information as shown in ECDSAOwnsership Module.

**2. Implement validateUserOp method**
```